### PR TITLE
Remove CurrentAdjacentCells update on each tick in Cargo

### PR DESCRIPTION
**Pull request by teinarss**
Just some cleanup in Cargo trait. Changed how the CurrentAdjacentCells was updated, from Tick() to on demand with cache.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Cargo.cs [hunk 1]
```diff
@@ -301,14 +306,6 @@ void ReleaseLock(Actor self)
 			takeOffAfterLoad = false;
 		}
 
-		public string CursorForOrder(Actor self, Order order)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Cargo.cs]: How is this related to the adjacent cells?

Comment (teinarss) [reply OpenRA.Mods.Common/Traits/Cargo.cs]: Not at all! It was a early Christmas present!

File: OpenRA.Mods.Common/Traits/Cargo.cs [hunk 2]
```diff
@@ -301,14 +306,6 @@ void ReleaseLock(Actor self)
...
-		{
-			if (order.OrderString != "Unload")
-				return null;
-
-			return CanUnload() ? Info.UnloadCursor : Info.UnloadBlockedCursor;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Cargo.cs]: Aren't those unused now?

Comment (teinarss) [reply OpenRA.Mods.Common/Traits/Cargo.cs]: No, they are still used https://github.com/OpenRA/OpenRA/pull/17376/files#diff-37ac89c3ba25e755edbe82852d067d49R204

**Comment by teinarss** [comment]
Updated and added another commit which removed the use of Tick

File: OpenRA.Mods.Common/Traits/Cargo.cs [hunk 3]
```diff
@@ -151,6 +161,17 @@ public Cargo(ActorInitializer init, CargoInfo info)
 				totalWeight = cargo.Sum(c => GetWeight(c));
 			}
 
+			foreach (var c in cargo)
+			{
+				c.Trait<Passenger>().Transport = self;
+
+				foreach (var nec in c.TraitsImplementing<INotifyEnteredCargo>())
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Cargo.cs]: Are we sure it is a good idea to make these calls before `Created` ran? (E.g. this might/wll break with https://github.com/OpenRA/OpenRA/blob/ae34410c8077a45d3dae20419aa63659cf0718a6/OpenRA.Mods.Common/Traits/Passenger.cs#L75-L78 and https://github.com/OpenRA/OpenRA/blob/ae34410c8077a45d3dae20419aa63659cf0718a6/OpenRA.Mods.Common/Traits/Passenger.cs#L123-L126 won't it?)

**Review by abcdefg30 — COMMENTED** [review]
Looks good otherwise.

File: OpenRA.Mods.Common/Traits/Cargo.cs (hunk 3 above)
Comment (teinarss) [reply OpenRA.Mods.Common/Traits/Cargo.cs]: Dont think so, INotifyCreated.Created will be called in the call chain `World.CreateActor() -> Actor.Created()` Right?
https://github.com/OpenRA/OpenRA/blob/1027b125e13e8efa47dddd7492dad71d8683cbf0/OpenRA.Mods.Common/Traits/Cargo.cs#L143-L144
or
https://github.com/OpenRA/OpenRA/blob/1027b125e13e8efa47dddd7492dad71d8683cbf0/OpenRA.Mods.Common/Traits/Cargo.cs#L155-L156

Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/Cargo.cs]: Ah yes, you're right, they probably run for each actor and not once all actors are done.

**Review by abcdefg30 — APPROVED** [review]

**Review by tovl — APPROVED** [review]

**Comment by tovl** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/99af84d1aca0059ba9eef930ecece4d4205367b5...2afd4d0a25051df342595c5f92c53dc30b3eba7f
